src/psyclone/tests/psyGen_test.py

```diff
@@ -2617,8 +2627,269 @@ def test_binaryoperation_can_be_printed():
+    rhs = Literal("1", parent=assignment)
+    assignment.addchild(lhs)
+    assignment.addchild(rhs)
+    kschedule.addchild(assignment)
+    assert "Schedule[name:'kname']:\n" in str(kschedule)
+    assert "Assignment" in str(kschedule)  # Check childs are printed
+    assert "End Schedule" in str(kschedule)
+
+
+# Test Symbol Class
+
+def test_symbol_can_be_printed():
+    symbol = Symbol("sname", "real", 1, "read_arg")
+    assert "sname<real,1,read_arg>" in str(symbol)
+
+
+# Test SymbolTable Class
+
+def test_symboltable_declare():
+    sym_table = SymbolTable()
+
+    # Declare a symbol
+    sym_table.declare("var1", "real", 3, "read_arg")
+    assert sym_table._symbols["var1"].name == "var1"
+    assert sym_table._symbols["var1"].datatype == "real"
+    assert sym_table._symbols["var1"].dimensions == 3
+    assert sym_table._symbols["var1"].kind == "read_arg"
+
+    # Declare a duplicate name symbol
+    with pytest.raises(InternalError) as error:
+        sym_table.declare("var1", "real", 0, "write_arg")
+    assert "Multiple definition of symbol: 'var1'." in str(error.value)
+
+
+def test_symboltable_lookup():
+    sym_table = SymbolTable()
+    sym_table.declare("var1", "real", 3, "read_arg")
+    sym_table.declare("var2", "integer", 0, "write_arg")
+    sym_table.declare("var3", "real", 0, None)
+
+    assert isinstance(sym_table.lookup("var1"), Symbol)
+    assert sym_table.lookup("var1").name == "var1"
+    assert isinstance(sym_table.lookup("var2"), Symbol)
+    assert sym_table.lookup("var2").name == "var2"
+    assert isinstance(sym_table.lookup("var3"), Symbol)
+    assert sym_table.lookup("var3").name == "var3"
+
+    with pytest.raises(KeyError) as error:
+        sym_table.lookup("notdeclared")
+
+
+def test_symboltable_view(capsys):
+    sym_table = SymbolTable()
+    sym_table.declare("var1", "real", 3, "read_arg")
+    sym_table.declare("var2", "integer", 2, "read_arg")
+    sym_table.view()
+    output, _ = capsys.readouterr()
+    assert "Symbol Table:\n" in output
+    assert "var1" in output
+    assert "var2" in output
+
+
+def test_symboltable_can_be_printed():
+    sym_table = SymbolTable()
+    sym_table.declare("var1", "real", 3, "read_arg")
+    sym_table.declare("var2", "integer", 2, "read_arg")
+    assert "Symbol Table:\n" in str(sym_table)
+    assert "var1" in str(sym_table)
+    assert "var2" in str(sym_table)
+
+
 # Test Fparser2ASTProcessor
 
+def test_fparser2astprocessor_generate_schedule():
+    ''' Tests the fparser2AST generate_schedule method.
+    '''
+    ast1 = fpapi.parse(FAKE_KERNEL_METADATA, ignore_comments=True)
+    metadata = DynKernMetadata(ast1)
+    my_kern = DynKern()
+    my_kern.load_meta(metadata)
+    ast2 = my_kern.ast
+    processor = Fparser2ASTProcessor()
+
+    # Test properly formed but empty kernel module
+    schedule = processor.generate_schedule("dummy_code", ast2)
+    assert isinstance(schedule, KernelSchedule)
+
+    # Test generate an unexistent subroutine name
```

Comment:
s/Test generate an unexistant/Test that we get an error for a nonexistant/

Reply:
Done